fix: Added Pitstops, Qualifying, Circuits, Constructors, Drivers Standings and Constructors standings endpoint docs (#83)
jolpica/jolpica-f1: docs/endpoints/races.md

[← Documentation Home](/docs/README.md)
# Races

> Gets list of races from earliest to latest.

Reviewer: ```suggestion
Returns a list of races from earliest to latest.
```

**URL** : `/ergast/f1/races/`

[Available Query Parameters](/docs/README.md#query-parameters)

---

## Route Parameters:

### Season

> **Filters for races only from a specified season. Year numbers are valid as is `current` to get the current season**

Reviewer: ```suggestion
Filters for races only from a specified season. Year numbers are valid as is `current` to get the current season.
```

`/{season}/` -> ex: `/ergast/f1/2024/races/`

> **Note**: To utilize the `season` parameter it needs to be the first route after `/ergast/f1/`

Reviewer: ```suggestion
**Note**: To utilize the `season` parameter it needs to be the first argument after `/ergast/f1/`.
```

---

### Round

> **Filters for the race for a specified round in a specific season. Round numbers 1 -> n races are valid as well as `last` and `next`**

Reviewer: ```suggestion
Filters for the race for a specified round in a specific season. Round numbers 1 -> `n` races are valid as well as `last` and `next`.
```

`/{season}/{round}/` -> ex: `/ergast/f1/2024/5/races/`

> **Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first route after `/ergast/f1/{season}/`

Reviewer: ```suggestion
**Note**: To utilize the `round` parameter it must be combined with a season filter and needs to be the first argument after `/ergast/f1/{season}/`.
```

---

### circuits

> **Filters for only races featuring a specified circuit**

Reviewer: ```suggestion
Filters for only races featuring a specified circuit.
```

`/circuits/{circuitId}/` -> ex: `/ergast/f1/circuits/monza/races/`

---

### constructors

> **Filters for only races featuring a specified constructor**

Reviewer: ```suggestion
Filters for only races featuring a specified constructor.
```

`/constructors/{constructorId}/` -> ex: `/ergast/f1/constructors/williams/races/`

---

### drivers

> **Filters for only races featuring a specified driver**

Reviewer: ```suggestion
Filters for only races featuring a specified driver.
```

**Filters for only races featuring a specified grid position**

`/grid/{gridPosition}/` -> ex: `/ergast/f1/grid/27/races/`

---

### status

> **Filters for only races featuring a specified finishing status of a driver**

Reviewer: ```suggestion
Filters for only races featuring a specified finishing status of a driver.
```

`MRData.RaceTable.Races` : The list of all races returned.

`MRData.RaceTable.Races[i]` : A given race object.

---

## Race Object Fields:

|Field|Always Included|Description|type
|---|:---:|---|---|
|season|✅|Season year|String
|round|✅|Round Number|String
|url|✅|Wikipedia URL of race|String
|raceName|✅|Name of the race|String
|Circuit|✅|Circuit information (circuitId, url, circuitName, Location)|Object
|date|✅|Date of the race (YYYY-MM-DD)|String
|time|🟡|UTC start time of the race|String
|FirstPractice|🟡|First Practice (date, time)|Object
|SecondPractice|🟡|Second Practice (date, time)|Object
|ThirdPractice|🟡|Third Practice (date, time)|Object
|Qualifying|🟡|Qualifying (date, time)|Object
|Sprint|🟡|Sprint Race (date, time)|Object

> |SprintQualifying|🟡|Sprint Qualifying (date, time)|Object

Reviewer 2: Also can have a `SprintShootout` field (I believe 2023).

> |SprintQualifying|🟡|Sprint Qualifying (date, time)|Object

Reviewer 2: ```suggestion
|SprintQualifying / SprintShootout|🟡| Shootouts took place in 2023, otherwise they are Qualifying (date, time)|Object
```

## Race Object Fields:

|Field|Always Included|
|---|:---:|
|season| ✅ |
|round|✅|
|url|✅|
|raceName|✅|
|Circuit|✅|
|date|✅|
|time|🟡|
|FirstPractice|🟡|
|SecondPractice|🟡|
|ThirdPractice|🟡|
|Qualifying|🟡|
|Sprint|🟡|
|SprintQualifying|🟡|

---

## Examples:

### Get list of races in F1 history

> `https://api.jolpi.ca/ergast/f1/races//`

Reviewer: ```suggestion
`https://api.jolpi.ca/ergast/f1/races/`
```
